dnase/callsnpsMerge.sh

```diff
@@ -217,6 +253,65 @@ for sampleid in `bcftools query -l ${sampleOutdir}/${name}.${mappedgenome}.filte
     #https://genome.ucsc.edu/FAQ/FAQformat.html#format10
 done
 
+echo
+echo "Parsing Delly VCF track"
+date
+#No additional filtering, just extract genotypes to starch
+#NB repeated from perChrom analysis
+minSNPQ=0
+#
+minTotalDP=0
+minAlleleDP=0
+
+${src}/parseSamtoolsGenotypesToBedFiles.pl ${sampleOutdir}/${name}.${mappedgenome}.delly.filtered.vcf.gz $TMPDIR/delly_variants ${minSNPQ} ${minTotalDP} ${minAlleleDP}
+
+nvcfsamples=`bcftools query -l ${sampleOutdir}/${name}.${mappedgenome}.delly.filtered.vcf.gz | wc -l`
+#rsids
+for sampleid in `bcftools query -l ${sampleOutdir}/${name}.${mappedgenome}.delly.filtered.vcf.gz`; do
+    if [[ "${nvcfsamples}" = 1 ]]; then
+        vcfsamplename=""
+    else
+        #for multisample calling, include the sample name in the variants file name
+        vcfsamplename=".${sampleid}"
+    fi
+    #If there is no database ID (e.g. rsid), set up bed ID as chrom : pos _ alt
+    cat $TMPDIR/delly_variants.${sampleid}.txt | awk -F "\t" 'BEGIN {OFS="\t"} $4=="." {split($8, gt, "/"); gtout=gt[1]; if(length(gt)>2) {gtout=gtout "/" gt[2]}; chromnum=$1; gsub(/^chr/, "", chromnum); $4=chromnum ":" $2+1 "_" gtout } {print}' | sort-bed - | starch - > ${sampleOutdir}/${name}${vcfsamplename}.${mappedgenome}.delly.genotypes.starch
+    
+    #NB UCSC link from analysis.sh will be wrong for multisample calling
+    #Start from .txt file to simplify logic even though we have to sort again
+    #If there is no database ID (e.g. rsid), set up bed ID as chrom : pos _ alt; for database IDs, append alt allele
+    #truncgt() is a fancy wrapper around substr for syntactic sugar and to add a "..."
+    cat $TMPDIR/delly_variants.${sampleid}.txt | awk -v maxlen=115 -F "\t" 'BEGIN {OFS="\t"} function truncgt(x) {if(length(x)>maxlen) {return substr(x, 1, maxlen) "..." } else {return x} } $4=="." {chromnum=$1; gsub(/^chr/, "", chromnum); $4=chromnum ":" $2+1 } {split($8, gt, "/"); gtout=truncgt(gt[1]); if(length(gt)>2) {gtout=gtout "/" truncgt(gt[2])}; $4=$4 "_" gtout } {print}' |
+    awk -F "\t" 'BEGIN {OFS="\t"} {
+        ref=$6;
+        numAlleles=split($8, gt, "/");
+        # Assign colors to the genotypes.bb lines.
+        if (numAlleles>1 && gt[1]!=gt[2]) {
+            color="19,165,220";        # het - blue
+        } else {
+            if (gt[1] == ref) {
+                color="105,105,105";   # hz_ref - grey
+            } else {
+                color="253,199,0";     # hz_nonref - yellow
+            }
+        }
+        # Delly has scores set to 10_000, this is outside of the 0->1_000 range
+        # and these values need to be clamped
+        score=$5;
+        if (score>1000) {
+          score=1000;
+        }
```

Comment:
this is safe to merge in with the original code, UCSC does not permit scores over 1000